### PR TITLE
chore: migrate VS projects

### PR DESCRIPTION
**Pull request by blackchoey**

### CODE REVIEW

File: packages/fx-core/templates/core/v3Migration/csharp.app.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
+version: 1.0.0
```
Comment (blackchoey) [inline packages/fx-core/templates/core/v3Migration/csharp.app.yml:1]: @IvanJobs @Siglud please help review the content in this app yml template, the syntax is handlebars syntax (similar to mustache)

**Review by Siglud — APPROVED** [review]

File: packages/fx-core/src/core/middleware/projectMigratorV3.ts [hunk 2]
```diff
@@ -192,12 +195,21 @@ export async function generateSettingsJson(context: MigrationContext): Promise<v
...
 
 export async function generateAppYml(context: MigrationContext): Promise<void> {
+  const inputs: Inputs = context.arguments[context.arguments.length - 1];
+  const bicepFilePath =
+    inputs.platform === Platform.VS
+      ? Constants.vsProvisionBicepPath
+      : Constants.vscodeProvisionBicepPath;
   const bicepContent: string = await fs.readFile(
-    path.join(context.projectPath, Constants.provisionBicepPath),
+    path.join(context.projectPath, bicepFilePath),
```
Comment (frankqianms) [inline packages/fx-core/src/core/middleware/projectMigratorV3.ts:204]: so this function: [readBicepContent](https://github.com/OfficeDev/TeamsFx/blob/dev/packages/fx-core/src/core/middleware/utils/v3MigrationUtils.ts#:~:text=export%20function%20readBicepContent(context%3A%20MigrationContext)%3A%20any%20%7B) should also be changed according to platform, right?

Comment (frankqianms) [reply packages/fx-core/src/core/middleware/projectMigratorV3.ts:204]: or could you please change the code of readBicepContent() in utils, as this is a common function that will be often refered?

File: packages/fx-core/templates/core/v3Migration/csharp.app.yml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+provision:
+{{#if activePlugins.fx-resource-bot}}
+  - uses: botAadApp/create # Creates a new AAD app for Bot Registration.
+    with:
+      name: {{appName}}bt$\{{RESOURCE_SUFFIX}}
```
Comment (IvanJobs) [inline packages/fx-core/templates/core/v3Migration/csharp.app.yml:7]: Nit: Just FYI, I saw multiple `\` used across the file content. Another way to make it clean is to use a different separator like this file: https://github.com/OfficeDev/TeamsFx/blob/dev/packages/fx-core/templates/plugins/resource/cicd_v2/github/cd.yml

Comment (blackchoey) [reply packages/fx-core/templates/core/v3Migration/csharp.app.yml:7]: thanks, I will have a try later

**Review by IvanJobs — DISMISSED** [review]
LGTM for the part of `botAadApp/create` in `csharp.app.yml`.

File: packages/fx-core/src/core/middleware/projectMigratorV3.ts (hunk 2 above)
Comment (blackchoey) [reply packages/fx-core/src/core/middleware/projectMigratorV3.ts]: thanks, updated

**Review by chagong — DISMISSED** [review]

**Review by frankqianms — DISMISSED** [review]

**Review by xzf0587 — APPROVED** [review]

**Pull request merged**